server/handlers/repos/unlink.go

```diff
@@ -0,0 +1,103 @@
+package repos
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/hn275/envhub/server/api"
+	"github.com/hn275/envhub/server/database"
+	"github.com/hn275/envhub/server/gh"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5/pgconn"
+)
+
+// returns 201 on success, no body
+// { "message": "err" } otherwise
+func Unlink(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodDelete {
+		api.NewResponse(w).Status(http.StatusMethodNotAllowed).Done()
+		return
+	}
+
+	user, err := api.NewContext(r).User()
+	if err != nil {
+		api.NewResponse(w).Status(http.StatusForbidden).Error(err.Error())
+		return
+	}
+
+	// only `repo.ID` and `repo.FullName` are sent
+	var repo database.Repository
+	var partialData map[string]interface{}
+
+	// Decode the JSON into a map
+	if err := json.NewDecoder(r.Body).Decode(&partialData); err != nil {
+		api.NewResponse(w).Status(http.StatusBadRequest).Error(err.Error())
+		return
+	}
+
+	// Populate the specific fields in the 'repo' struct
+	if id, ok := partialData["repoID"].(uint64); ok {
+		repo.ID = id
+	}
+	if fullName, ok := partialData["repoName"].(string); ok {
+		repo.FullName = fullName
+	}
+
+	// Now, 'repo' should contain the decoded values
+	// log.Println("Received repo:", repo)
+
+	// CHECK FOR REPO OWNER
+	// get repoInfo
+	var repoInfo Repository
+	// https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository
+	res, err := gh.New(user.Token).Get("/repos/%s", repo.FullName)
```

Comment:
I'm not sure we need to make an external api call even, the `Repository` struct has an `UserID` field, we may just need to query db for it then delete the repo if the user id of the query matches with the user id of the request.